EmbeddedSocial/Sources/Model/Services/Comments/CommentsService.swift

```diff
@@ -261,10 +261,18 @@ class CommentsService: BaseService, CommentServiceProtocol {
         
         let request = CommentsAPI.commentsDeleteCommentWithRequestBuilder(commentHandle: command.comment.commentHandle, authorization: authorization)
         request.execute { (response, error) in
-            if let error = error {
-                failure(error)
-            } else {
+            if let response = response {
                 success()
+            } else if self.errorHandler.canHandle(error) {
+                self.errorHandler.handle(error)
+                failure(APIError(error: error))
+            } else {
+                if error!.statusCode > Constants.HTTPStatusCodes.InternalServerError.rawValue {
+                    self.cache.cacheOutgoing(command)
+                    failure(APIError(error: error))
```

Comment:
Error 500 is considered a successful response. Please change the callback into `success`.